### PR TITLE
Add Javadoc (part 1) and extra fixes

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Add missing Javadoc reported by the checkstyle plugin.
Will selectively add more Javadoc in important/complex code paths in a different PR.

### Extra Fixes

- Remove unused parameter in `MonitorConnectionContext`
- Fix a missing change lost during rebase.

### Additional Reviewers

@sergiyv-bitquill 
@ColinKYuen 
@matthewh-BQ

### CODE REVIEW

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java [hunk 1]
```diff
@@ -58,6 +58,9 @@
     Runtime.getRuntime().addShutdownHook(new Thread(ConnectionPluginManager::releaseAllResources));
   }
 
+  /**
+   * Constructor.
```
Comment (ColinKYuen) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java:62]: Include the parameters

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java [hunk 2]
```diff
@@ -48,6 +52,11 @@
   private final ExecutorService threadPool;
   private static final Object LOCK_OBJECT = new Object();
 
+  /**
+   * Create an instance of the {@link MonitorThreadContainer}.
+   *
+   * @return an instance of
```
Comment (ColinKYuen) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java:58]: an instance of...

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java [hunk 3]
```diff
@@ -63,6 +72,10 @@ static MonitorThreadContainer getInstance(IExecutorServiceInitializer executorSe
     return singleton;
   }
 
+  /**
+   * Release resources held in the the {@link MonitorThreadContainer} and clear reference
```
Comment (ColinKYuen) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java:76]: the the

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java (hunk 1 above)
Comment (karenc-bq) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java:62]: Excluding parameters for constructors that only require loggers. I am tempted to remove these single line javadoc too as it does not provide value.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java [hunk 4]
```diff
@@ -43,6 +43,10 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 
+/**
+ * This class uses a background thread to monitor each server with one or more active
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java:47]: each server -> particular server, concrete server ?

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorConnectionContext.java [hunk 5]
```diff
@@ -47,6 +51,17 @@
...
   private boolean activeContext = true;
 
+  /**
+   * Constructor.
+   *
+   * @param connectionToAbort A reference to the connection associated with this context
+   *                          that will be aborted in case of server failure.
+   * @param nodeKeys All valid references to the server.
+   * @param log A {@link Log} implementation.
+   * @param failureDetectionTimeMillis Grace period during which failures are ignored.
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorConnectionContext.java:61]: Grace period after which node monitoring is starting ?

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringConnectionPlugin.java [hunk 6]
```diff
@@ -45,9 +45,14 @@
 import java.util.concurrent.Callable;
 import java.util.function.Supplier;
 
+/**
+ * Monitor the server while the connection is executing methods for more flexible
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringConnectionPlugin.java:49]: flexible -> sophisticated?

**Review by sergiyv-bitquill — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

**Pull request merged**